### PR TITLE
feat: add sauce performance and extended debugging options (#118)

### LINKED ISSUE #118: Add additional sauce desired capabilities for performance metrics
# Summary
Sauce labs offers performance metric tracking. It looks like they provide sauce options to enable the metrics, [found on this doc here](https://docs.saucelabs.com/performance/transitions/).

We should add these sauce options to our sauce desired capabilities, [here.](https://github.com/kgress/scaffold/blob/master/environment/src/main/java/io/github/kgress/scaffold/environment/config/DesiredCapabilitiesConfigurationProperties.java#L128). We should also check to see if there are any other sauce options we'd like to add, which can be found from [this document here](https://docs.saucelabs.com/dev/test-configuration-options/#desktop-browser-capabilities-sauce-specific--optional). 

These options should be checked for on the `WebDriverManager`'s `configureSauceBrowserOptions()` method, [found here](https://github.com/kgress/scaffold/blob/master/framework/src/main/java/io/github/kgress/scaffold/WebDriverManager.java#L282).

# A/C
* Scaffold should provide the sauce options, "capturePerformance" and "extendedDebugging."

### PR DESCRIPTION
**Pull request by atari-xia**
- adds capturePerformance and extendedDebugging options to DesiredCapabilities for sauce

closes #118

### CODE REVIEW

File: environment/src/main/java/io/github/kgress/scaffold/environment/config/DesiredCapabilitiesConfigurationProperties.java [hunk 1]
```diff
@@ -167,6 +167,18 @@
...
+     * This enables the extended debugging feature on SauceLabs. The default on SauceLabs is to have this set as false.
+     * Required to be true for capturePerformance to be enabled.
+     */
+    private boolean extendedDebugging = false;
+
+    /**
+     * This enables the performance capturing feature on SauceLabs. The default on SauceLabs is to have this set as false.
+     * extendedDebugging must also be true for this to work.
+     */
+    private boolean capturePerformance = false;
```
Comment (kgress) [inline environment/src/main/java/io/github/kgress/scaffold/environment/config/DesiredCapabilitiesConfigurationProperties.java:181]: On usage with SauceLabs: if these values are just not present in the request, does SauceLabs automatically default it as `false`?

If that's the case, I think we should remove the default value being set here as `false`, and then update the `WebDriverManager` to build `Optional.ofNullable()` values, check if these values are present, and set them as sauce capabilities. That logic [is found here](https://github.com/kgress/scaffold/blob/551e6d151e4c22aeef980e5b7637dc4e0424b94f/framework/src/main/java/io/github/kgress/scaffold/WebDriverManager.java#L607). These two values would be set just under like 610.

File: framework/src/main/java/io/github/kgress/scaffold/WebDriverManager.java [hunk 2]
```diff
@@ -588,6 +588,8 @@ private RemoteWebDriver configureSauceRemoteBrowser(MutableCapabilities browserO
       sauceCaps.setCapability("username", username);
       sauceCaps.setCapability("accessKey", accessKey);
       sauceCaps.setCapability("name", testName);
+      sauceCaps.setCapability("extendedDebugging", sauce.isExtendedDebugging());
+      sauceCaps.setCapability("capturePerformance", sauce.isCapturePerformance());
```
Comment (kgress) [inline framework/src/main/java/io/github/kgress/scaffold/WebDriverManager.java:592]: Comment on `DesiredCapabiliitiesConfigurationProperties` will impact this addition slightly, moving it to just under line 610 with the `Optional`'s `.isPresent()` check.

**Review by kgress — CHANGES_REQUESTED** [review]
Thanks for the PR submission on this! Just a small tweak to the defaults and how it's being set, and we can merge

File: environment/src/main/java/io/github/kgress/scaffold/environment/config/DesiredCapabilitiesConfigurationProperties.java (hunk 1 above)
Comment (atari-xia) [reply environment/src/main/java/io/github/kgress/scaffold/environment/config/DesiredCapabilitiesConfigurationProperties.java]: Saucelabs defaults the values to false. I've made the changes to not send these values if they are unset.

File: environment/src/main/java/io/github/kgress/scaffold/environment/config/DesiredCapabilitiesConfigurationProperties.java [hunk 3]
```diff
@@ -167,6 +167,18 @@
...
+     * This enables the extended debugging feature on SauceLabs. The default on SauceLabs is to have this set as false.
+     * Required to be true for capturePerformance to be enabled.
+     */
+    private Boolean extendedDebugging;
+
+    /**
+     * This enables the performance capturing feature on SauceLabs. The default on SauceLabs is to have this set as false.
+     * extendedDebugging must also be true for this to work.
+     */
+    private Boolean capturePerformance;
```
[the remark this answers is not in the document]
Comment (kgress) [reply environment/src/main/java/io/github/kgress/scaffold/environment/config/DesiredCapabilitiesConfigurationProperties.java:181]: ignore sonatype lift, disabling.

File: environment/src/main/java/io/github/kgress/scaffold/environment/config/DesiredCapabilitiesConfigurationProperties.java [hunk 4]
```diff
@@ -167,6 +167,18 @@
      * Optional. The custom time zone to use when spinning up the sauce lab VM.
      */
     private String timeZone;
+
+    /**
+     * This enables the extended debugging feature on SauceLabs. The default on SauceLabs is to have this set as false.
+     * Required to be true for capturePerformance to be enabled.
+     */
+    private Boolean extendedDebugging;
```
[the remark this answers is not in the document]
Comment (kgress) [reply environment/src/main/java/io/github/kgress/scaffold/environment/config/DesiredCapabilitiesConfigurationProperties.java:175]: ignore sonatype lift, disabling.

**Review by kgress — APPROVED** [review]

**Pull request merged**